test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -1211,10 +1214,18 @@ runIO action lookUp = ReaderT $ \ !env -> do
           Class.mupserts (unwrapTable $ lookUp' tableVar) kmups
         RetrieveBlobs blobRefsVar -> catchErr handlers $
           fmap WrapBlob <$> Class.retrieveBlobs (Proxy @h) session (unwrapBlobRef <$> lookUp' blobRefsVar)
-        CreateSnapshot merrs label name tableVar ->
-          runRealWithInjectedErrors "CreateSnapshot" env merrs
-            (Class.createSnapshot label name (unwrapTable $ lookUp' tableVar))
-            (\() -> Class.deleteSnapshot session name)
+        CreateSnapshot mcorrsOrErrs label name tableVar ->
+          let rCreateSnapshot = Class.createSnapshot label name (unwrapTable $ lookUp' tableVar) in
+          case sequence mcorrsOrErrs of
+            Left (SilentCorruptions corrs) -> do
+              rCreateSnapshot
+              for_ corrs $ \corr ->
+                Class.corruptSnapshot (bitChoice corr) name (unwrapTable $ lookUp' tableVar)
```

Comment:
We could technically flip a bit we previously flipped, un-corrupting the snapshot again. It's verrrry unlikely, but it might deserve a comment. What do you think?

Reply:
Ugh, yes. Fine, I'll `Set` it! ^__^"